lib/inspec/profile.rb

```diff
@@ -211,6 +212,32 @@ def locked_dependencies
       @locked_dependencies ||= load_dependencies
     end
 
+    def lockfile_exists?
+      File.exist?(lockfile_path)
+    end
+
+    def lockfile_path
+      File.join(cwd, 'inspec.lock')
+    end
+
+    def cwd
+      File.directory?(@target) ? @target : nil
+    end
+
+    def lockfile
+      @lockfile ||= if lockfile_exists?
+                      Inspec::Lockfile.from_file(lockfile_path)
+                    else
+                      generate_lockfile
+                    end
+    end
+
+    def generate_lockfile(vendor_path = nil)
```

Comment:
Comment: Make this an in-memory operation that doesn't render the lock file to disk.